### DISCUSSION TITLE
Add minification to PyScript's Python code (The included `pyscript` namespace)

### DISCUSSION THREAD

**Discussion by ntoll** [pyscript/pyscript#2139] [Proposals]
After a quick exploration, [python-minifier](https://github.com/dflook/python-minifier) looks a good tool to achieve this end.

```bash
$ pip install python-minifier
$ pyminify web.py > web_minified.py
$ ls -la web*
-rw-rw-r-- 1 ntoll ntoll 26003 Aug  6 16:06 web_minified.py
-rw-rw-r-- 1 ntoll ntoll 36702 Aug  6 16:05 web.py
```

Docs here: https://dflook.github.io/python-minifier/

The tool uses techniques like replacing usual white-space (4 spaces. 4 bytes) with a single tab (1 byte), removing comments, single-lining things if possible and other fun tricks.

For this feature to land we need the following four things to happen:

1. Ensure `python-minifier` is installed.
2. `pyminify` all the Python `stdlib` somewhere temporary.
3. Rollup the minified things rather than the "normal" source files.
4. Clean-up / ignore the minified assets automatically.

**Comment by WebReflection** [comment]
@ntoll I am incapable of installing this module ...

> python-minifier currently supports Python 2.7 and Python 3.3 to 3.11

I am on `Python 3.12.4` and I don't like the fact this module can't work on latest Python ... sounds like screaming for troubles moving forward, imho.

**Reply by WebReflection** [reply]
uhm ... OK, this worked apparently:

```sh
pip install --ignore-requires-python python-minifier
pip install setuptools
```

I will play around to see how we can hook this into our current build process.

**Reply by ntoll** [reply]
Yeah, I noticed this.

Then I found their issue for updating to 3.12 - they have a very conscientious checklist of things to work through so they remain compatible: https://github.com/dflook/python-minifier/pull/94

The maintainer has been working on this in the past 24hours, which also bodes well (i.e. it's carefully and actively maintained... but they're also a volunteer... so we have to respect that the fun things might not land immediately).

:+1:

**Reply by WebReflection** [reply]
https://github.com/pyscript/pyscript/pull/2140

**Comment by WebReflection** [comment]
I think we can close this as done.
